lib/store/settings_store.dart

```diff
@@ -135,22 +136,23 @@ abstract class SettingsStoreBase with Store {
         (bool shouldSaveRecipientAddress) => sharedPreferences.setBool(
             PreferencesKey.shouldSaveRecipientAddressKey, shouldSaveRecipientAddress));
 
-    setIsAppSecureNative(isAppSecure);
-
+    if (DeviceInfo.instance.isMobile) {
+      setIsAppSecureNative(isAppSecure);
+    }
     reaction((_) => isAppSecure, (bool isAppSecure) {
       sharedPreferences.setBool(PreferencesKey.isAppSecureKey, isAppSecure);
+      if (DeviceInfo.instance.isMobile) {
         setIsAppSecureNative(isAppSecure);
+      }
     });
```

Comment:
you haven't added this if condition on the UI too, thus the option will still be visible in Desktop settings, but it won't be working, we wanna remove it from the UI as well

also you can make group the function `setIsAppSecureNative(isAppSecure);` and the `reaction` in one if condition since they are related